Unit test fix setup_chains()

dsiganos opened the pull request:
Fixing a race condition in unit test function setup_chains()

dsiganos committed:
Minor compilation-time fix to EXPECT_TIMELY

EXPECT_TIMELY defined the variable "ec" without taking precautions
to restrict its scope. One symptom was that EXPECT_TIMELY could not be
used twice in a function.

dsiganos committed:
Fix race condition in setup_chains() in bootstrap_server.cpp

clemahieu commented on nano/test_common/testutil.hpp:
```diff
@@ -56,13 +56,15 @@
-	EXPECT_NO_ERROR (ec);
+#define EXPECT_TIMELY(time, condition)                  \
+	system.deadline_set (time);                         \
+	{                                                   \
```
Whats's the reason for the extra scope here?

dsiganos replied:
It is a C style macro that creates a local variable.
It is not ideal, so I limited its scope and added an underscore to the variable name.

pwojcikdev commented:
Every call to `nano::test::confirm()` should be done with ASSERT_TIMELY. I implemented that helper with intention to be used that way, I'm not sure why I didn't do that in other places.

dsiganos commented:
Hmmm, OK, I can see more calls to nano::test::confirm()  that do not use TIMELY asserts, I'll review/fix them.

dsiganos committed:
Fix possible race conditions in use of nano::test::confirm()

Most of these calls are race conditions because they use the following
pattern without allowing for propagation to occur:
process()
confirm()
